### ISSUE TITLE
CryptoMiniSat build script needs updating

### ISSUE THREAD

**Issue by gussmith23** [stp/stp#481]
CMS has added new dependencies. Their README now has the following installation script:

```sh
sudo apt-get install build-essential cmake

# not required but very useful
sudo apt-get install zlib1g-dev

git clone https://github.com/meelgroup/cadical
cd cadical
git checkout mate-only-libraries-1.8.0
./configure
make
cd ..

git clone https://github.com/meelgroup/cadiback
cd cadiback
git checkout mate
./configure
make
cd ..

tar xzvf cryptominisat-version.tar.gz
cd cryptominisat-version
mkdir build && cd build
cmake ..
make
sudo make install
sudo ldconfig
```

Specifically, it seems like the CadiBack dependency is missing.

Seems like there's two ways to fix:
- Build the CadiBack dependency in the following script: https://github.com/stp/stp/blob/master/scripts/deps/setup-cms.sh
- Allow the user to specify a specific CMS commit to be built when calling `setup-cms.sh`, which would allow us to use an old version pre-CadiBack dependency

Either way, should be a quick fix!

**Comment by aytey** [comment]
Thanks for the report; we'll take a look.

@msoos: is CaDiCaL now a required dependency for CMS? I'm just surprised that to build a SAT solver it seems you now need another SAT solver.

**Comment by msoos** [comment]
Hi,

Yeah, sorry. I didn't now where to integrate cadiback. I didn't want to reimplement cadiback... so yeah. Also, I wanted to give credit, where credit is due -- CadiBack. This is for [Arjun](https://github.com/meelgroup/arjun), the preprocessor that we have built. It's an insanely high-performance CNF preprocessor for model counting and also SAT solving. Unfortunately, it requires backbone detection. I have built one, and it's OK, but it's not like CadiBack. To be clear, I could improve CadiBack, if I had time, and I will improve it, but... not now. Don't know what to do. Sorry.

Mate
